src/main/lib/icon/IconFileStorage.js

```diff
@@ -19,13 +20,14 @@ export default class IconFileStorage extends StorageInterface {
     return path.join(this.path, fileName);
   }
 
-  /**
-     * Decode icon data that was pulled from storage
-     * @param {*} data The encoded data as it was stored
-     * @returns {Promise.<Buffer>} A promise that resolves with a buffer
-     */
-  decodeIconFromStorage(data) {
-    return Promise.resolve(data);
+  async encodeIconForStorage(iconData) {
+    const reader = new window.FileReader();
+    reader.readAsDataURL(iconData);
+    return new Promise(resolve => {
+      reader.onloadend = () => {
+        resolve(reader.result);
+      };
```

Comment:
Shouldn't you use rather `reader.onload` and `reader.onerror`, resolving and rejecting the Promise in each case?
Otherwise, it would just still store some bad result in case of failure (not even sure what would be the value), and would not retry anymore - when maybe retrying later to fetch the icon correctly might work and store the correct one.

If it rejects, it would cause Iconographer to reject, but that would not be a problem as we catch in `entries` and log if necessary.